### DISCUSSION TITLE
Organization > Team > Invite Member Dialog validates on blur, causing jarring form UX

### DISCUSSION THREAD

**Discussion by brandongregoryscott** [supabase/supabase#16925] [Contribute to Supabase]
# Bug report

## Describe the bug

The dialog for inviting members to an organization (on the hosted / app.supabase.com instance) is a bit jarring because the required email field validates immediately. If you click on the dropdown to select a role, or if you simply click off of the email input that is auto-focused when you open the dialog, it will mark the field as invalid. 

I dug into the code a bit and it seems like this may be intentional, or at least it's *consistent* across components that leverage this `Form` component and a validator: https://github.com/supabase/supabase/blob/ab9639064639023768a3479d0703c0921d03cad4/packages/ui/src/components/Form/Form.tsx#L41

How does the team feel about this? It feels like it should only update the validation state once the call-to-action button is clicked.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. After logging in, click on an organization on the left sidebar
1. Click on the 'Team' tab to visit the team management page
1. Click on the 'Invite' button to open the dialog
1. Click on the role dropdown OR click in the blank space in the dialog
1. Notice the email field immediately change to invalid

## Expected behavior

Since I have not yet completed the form, I would not expect validation errors to show up until I click the call-to-action button.

## Screenshots

![invite-member-button](https://user-images.githubusercontent.com/11774799/224186923-7c82c100-efd2-4415-a1c1-83283ff89b09.gif)

## System information

- OS: macOS 12.3
- Browser (if applies) Chrome 110.0.5481.177 (Official Build) (arm64) 
- Version of supabase-js: N/A
- Version of Node.js: N/A

## Additional context

N/A

**Comment by kunimei** [comment]
thankyou
